### PR TITLE
Jobs and training content 

### PR DESCRIPTION
**Pull request by hiba-machfej**
I added the page content, I will do the search functionality in a seperate PR.

### CODE REVIEW

**Comment by hiba-machfej** [comment]
This is how the PR looks now :
Screen Shot 2020-11-13 at 3 35 03 PM
I added a function to disable the selected button. because without it, when the page renders both buttons would be styled plain and did not give any impression of which button is selected/which category are we reading.

File: src/components/JobsTrainingContent/index.jsx
Comment (Jamil-Bailony) [inline src/components/JobsTrainingContent/index.jsx]: why didn't you move this to `selectButton` function? doing that would help you write this in one line like:
```
onClick={selectButton}
```
or a different name other that `selectButton` as I will hold another functionality

**Review by Jamil-Bailony — APPROVED** [review]
Nothing other than that, thank you very much

File: src/components/JobsTrainingContent/index.jsx
Comment (Jamil-Bailony) [inline src/components/JobsTrainingContent/index.jsx]: I have a note about changing cards color in each row, I think it would so nice if you can change the colors in each row not making each column with one color, for example:
```
peach green
green peach
peach green
.... ....
```
as I remember, our UI/UX supervisor gave us the same note about it.

File: src/components/JobsTrainingContent/index.jsx
Comment (louisrli) [inline src/components/JobsTrainingContent/index.jsx]: technically the variable on line 12 is also a cardsArr. so you want a more precise variable here to express what this is. otherwise when i see something called "cardsArr" followed by something called "cards" i ask myself, is the second one not also an arr?

File: src/components/JobsTrainingContent/index.jsx
Comment (louisrli) [inline src/components/JobsTrainingContent/index.jsx]: keep hooks at top of component
https://reactjs.org/warnings/invalid-hook-call-warning.html#breaking-the-rules-of-hooks

File: src/components/JobsTrainingContent/index.jsx
Comment (louisrli) [inline src/components/JobsTrainingContent/index.jsx]: is this a real prop to an html button?

File: src/components/JobsTrainingContent/index.jsx
Comment (louisrli) [inline src/components/JobsTrainingContent/index.jsx]: this is another instance of copy-pasted code here if i told you, for example, you had to delete line 59 (which i think you do), you have to remember to do it in 2 places

a thing that happens a lot in reviewing this type of code:

me: delete line 59
them: ok
them: forgets to do it on line 46 (bc the code was copy-pasted. also look at the most recent comments on jamil's PR for an example of this happening)

but also, sometimes "me" is not louis but yourself, when you're making changes to your own code

basically, move the button into either a function `const opportunityButton(buttonText, translationKey)` or make it another component (slightly cleaner) (it can be in the same file, some people want only one comp. per file but i don't care esp. if the component is not used outside the file)

File: src/components/JobsTrainingContent/index.scss
Comment (louisrli) [inline src/components/JobsTrainingContent/index.scss]: can this just be one class, opportunityListBtn or something like this. unless you need to distinguish the CSS for trainings and jobs for some reason

File: src/components/JobsTrainingContent/index.jsx
Comment (hiba-machfej) [reply src/components/JobsTrainingContent/index.jsx]: ok I can change the name.

File: src/components/JobsTrainingContent/index.jsx
Comment (hiba-machfej) [reply src/components/JobsTrainingContent/index.jsx]: Actually, no the UI/UX designer wanted to keep all in one color. The thing you are suggesting is what we have in Figma right now. But I could not find away to do it. So, I did the PR like this for now because the functionality is more important. and I can discuss it with all of you.

File: src/components/JobsTrainingContent/index.jsx
Comment (hiba-machfej) [reply src/components/JobsTrainingContent/index.jsx:46]: ah, I think this should be type="button". actually, I fixed this in the search functionality branch and forgot to do it here. It also needs to be changed on the home page but I'll check with Jundi first as he wrote the code on the home page.

File: src/components/JobsTrainingContent/index.jsx
Comment (hiba-machfej) [reply src/components/JobsTrainingContent/index.jsx]: Actually, I took the code from the JobsTrainingHome so it used somewhere else. I will create another component so we can add to home page too 👍

File: src/components/JobsTrainingContent/index.scss
Comment (hiba-machfej) [reply src/components/JobsTrainingContent/index.scss]: It can be the same class.

File: src/components/JobsTrainingContent/index.jsx
Comment (hiba-machfej) [reply src/components/JobsTrainingContent/index.jsx]: I will check with Jundi since he also used it on the home page. I think translationKey="button" is a typo for  type="button" because on home page there is no type="button"

File: src/components/JobsTrainingContent/index.jsx
Comment (hiba-machfej) [reply src/components/JobsTrainingContent/index.jsx]: I actually used cards and ardsArr to match the JobsTrainingHome, I will change the naming.

File: src/components/JobsTrainingContent/index.jsx
Comment (hiba-machfej) [reply src/components/JobsTrainingContent/index.jsx]: @louisrli I did a separate PR for the button so this PR won't become so big. After merging I can push this code again

**Comment by hiba-machfej** [comment]
Search functionality is added also

File: src/components/JobsTrainingContent/index.jsx
Comment (louisrli) [inline src/components/JobsTrainingContent/index.jsx]: it's weird that there's two set calls right next to each other. is that right

File: src/components/JobsTrainingContent/index.jsx
Comment (louisrli) [inline src/components/JobsTrainingContent/index.jsx]: is a search card a search result or a card with a search on it? if the former, it needs a better name

File: src/containers/JobsTrainingPage/index.jsx
Comment (louisrli) [inline src/containers/JobsTrainingPage/index.jsx]: did you just add this in this PR?

**Review by louisrli — CHANGES_REQUESTED** [review]
can you rename this file to Opportunities instead of JobTraining

File: src/containers/JobsTrainingPage/index.jsx
Comment (hiba-machfej) [reply src/containers/JobsTrainingPage/index.jsx]: No it's from the old PR, I did not have search functionality.

**Comment by hiba-machfej** [comment]
I fixed the code. and I renamed the file. Should I rename other components from JobsTrainging into Opportunities in a separate PR ?

**Comment by hiba-machfej** [comment]
I think this code can't be merged until we merge the buttons because it depends on it.

**Comment by hiba-machfej** [comment]
this PR Closes #67

File: src/components/OpportunitiesContent/index.jsx
Comment (louisrli) [inline src/components/OpportunitiesContent/index.jsx]: (previousCards)=>

File: src/components/OpportunitiesContent/index.jsx
Comment (louisrli) [inline src/components/OpportunitiesContent/index.jsx]: you don't need this line, just filter on previousCards

**Review by louisrli — APPROVED** [review]

File: src/components/OpportunitiesContent/index.jsx
Comment (hiba-machfej) [reply src/components/OpportunitiesContent/index.jsx]: without it, if I search twice, second-time search doesn't appear because it would be searching the filtered array.

Comment (hiba-machfej) [reply src/components/OpportunitiesContent/index.jsx:18]: @louisli

Comment (hiba-machfej) [reply src/components/OpportunitiesContent/index.jsx]: @louisrli

Comment (louisrli) [reply src/components/OpportunitiesContent/index.jsx]: i'm a bit confused how searching twice works right now. i don't see anything that sets the cardsArr back to the default

* user types "foo"
* setCardsArr -> cardsArr cards contain foo
* user deletes "foo"
* ?? which code sets it back to the original default state of all cards?

Comment (hiba-machfej) [reply src/components/OpportunitiesContent/index.jsx]: Const searchedCards = [...cardsArr].

I think the handle search is filtering the variable searchedCards which contains all the elements of cardsArr before it's filtered.

Comment (louisrli) [reply src/components/OpportunitiesContent/index.jsx]: yes, but then you call setCardsArr, so cardsArr has now changed to your new set

i've written searchbox code like this a few times before so to my understanding you are going to need to keep a variable that keeps track of the original set of cards.

the original set of cards is kept in cardsArr. you call setCardsArr which changes cardsArr. therefore you should no longer have access to the original when you call `[...cardsArr]`. does that make sense? i'm only reading the code here, maybe it works for some reason

Comment (hiba-machfej) [reply src/components/OpportunitiesContent/index.jsx]: you are reading the code correctly, search twice is not working 😢

Comment (hiba-machfej) [reply src/components/OpportunitiesContent/index.jsx]: If I did a copy of the cardsArr like before:
```const [searchCard, setSearchCards] = useState(cardsArr);```
and in handle search I can reset the cards before filtering
 ```setSearchCards(cardsArr);```

but I'll have to pass it also to JobsTrainingButtons and set both of them to opportunityData
 ```setCardsArr(opportunityData);  ```
 ```setSearchCards(opportunityData) ```

It works like that. I just wanted to make sure, is it best practice or do you suggest is another way?

Comment (hiba-machfej) [reply src/components/OpportunitiesContent/index.jsx]: or should I PR what I am suggesting and then you can check?

Comment (louisrli) [reply src/components/OpportunitiesContent/index.jsx]: you can do something like that but they need better names. i'll leave it to you to think of it, but you should think about why "cardsarr" and "searchcards" are not expressing what you want them to express if they play those roles.

Comment (hiba-machfej) [reply src/components/OpportunitiesContent/index.jsx]: I guess searchCards is not good naming because it doesn't only give the wrong impression but it will be mapped and used even before we trigger search button.

So maybe it can be renderedCards ? Since it is related to the cards appearing on screen whether they are filtered or not.

I am not sure about cardsArr I'll think about why it is not correct naming. 
But as a first thought maybe it can be selectedOpportunityCards? Since it will always contain and toggle between opportunity data.

**Comment by louisrli** [comment]
generally right idea, but whatever the naming is, it has to make very very
clear that one of them is "the pool of all cards" and the other one is
"cards that were selected based on searching."

**Comment by hiba-machfej** [comment]
@louisrli is the naming better now?

File: src/components/OpportunitiesContent/index.jsx
Comment (louisrli) [inline src/components/OpportunitiesContent/index.jsx:12]: add a comment above here explaining the diff too

**Review by louisrli — APPROVED** [review]

File: src/components/OpportunitiesContent/index.jsx
Comment (hiba-machfej) [reply src/components/OpportunitiesContent/index.jsx:12]: done :D

**Pull request merged**
